src/main/java/org/broadinstitute/hellbender/engine/filters/ReadFilterLibrary.java

```diff
@@ -89,14 +100,36 @@ private ReadFilterLibrary(){ /*no instance*/ }
         @Override public boolean test(final GATKRead read){
             return ! read.isDuplicate();}}
 
+    public static class NotSecondaryAlignmentReadFilter extends ReadFilter {
+        private static final long serialVersionUID = 1L;
+        @Override public boolean test(final GATKRead read) {
+            return !read.isSecondaryAlignment();}}
+
+    public static class NotSupplementaryAlignmentReadFilter extends ReadFilter {
+        private static final long serialVersionUID = 1L;
+        @Override public boolean test(final GATKRead read) {
+            return !read.isSupplementaryAlignment();}}
+
+    public static class PairedReadFilter extends ReadFilter {
+        private static final long serialVersionUID = 1L;
+        @Override public boolean test(final GATKRead read) {
+            return read.isPaired();}}
+
     public static class PassesVendorQualityCheckReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
         @Override public boolean test(final GATKRead read){
             return ! read.failsVendorQualityCheck();}}
 
+    public static class ProperlyPairedReadFilter extends ReadFilter {
+        private static final long serialVersionUID = 1L;
+        @Override public boolean test(final GATKRead read) {
+            return read.isProperlyPaired();}}
+
     public static class PrimaryAlignmentReadFilter extends ReadFilter {
         private static final long serialVersionUID = 1L;
-        @Override public boolean test(final GATKRead read){
+        @Override public boolean test(final GATKRead read) {
+            //TODO: should this also be rejecting isSupplementaryAlignment() ?
```

Comment:
I created a ticket to resolve whether we should change this or delete it (since we now have NotSecondary for InsertSizeMetrics) https://github.com/broadinstitute/gatk/issues/2165.